timetable/utils.py

```diff
@@ -41,26 +41,26 @@ def get_timetable_context(lessons):
 
     lessons = lessons.select_related('teacher', 'group', 'room', 'subject')
 
+    # TODO: a cleaner way to pass period str to the template while using
+    #       period number as key?
     periods = [period.number for period in default_periods]
     period_strs = get_period_strings(default_periods)
 
-    alternative_periods = Period.objects.exclude(schedule=1) # Exclude default periods
-
-    # TODO: a cleaner way to pass period str to the template while using
-    #       period number as key?
-    alternative_periods = Period.objects.exclude(schedule=1) # Exclude default periods
-    alternative_day_plan = None
-    for day_plan in DayPlan.objects.all():
-        if day_plan.is_today:
-            alternative_day_plan = day_plan
-            break
+    try:
+        alternative_day_plan = DayPlan.objects.get(date=date.today())
+        alternative_periods = Period.objects.filter(schedule=alternative_day_plan.schedule)
+    except ObjectDoesNotExist:
+        alternative_day_plan = None
```

Comment:
Here you are catching `ObjectDoesNotExist`. In our case it should work _**perfectly fine**_, but for the future I strongly recommend using specified `Model.DoesNotExist` exception, because you have greater control over which model throws an exception. You can confidently use `ObjectDoesNotExist` exception when you're 100% sure which query will cause the exception. If somehow `Period.objects.filter()` method throws an exception (which I don't beileve it will, but still), your `alternative_day_plan` will be wrongly set to `None`. Additionaly, by using specified model exception, the code will be more readable and easy to understand for other devs.

Reply:
Thanks, I agree that this is clearer and easier to understand this way. I have changed it to catch `DayPlan.DoesNotExist` exception and also to catch `Period.DoesNotExist` exception bellow.